volatility3/framework/plugins/windows/strings.py

```diff
@@ -81,22 +87,45 @@ def _generator(self) -> Generator[Tuple, None, None]:
         last_prog: float = 0
         line_count: float = 0
         num_strings = len(string_list)
-        for offset, string in string_list:
+        for phys_offset, string in string_list:
             line_count += 1
-            try:
-                revmap_list = [
-                    name + ":" + hex(offset) for (name, offset) in revmap[offset >> 12]
-                ]
-            except (IndexError, KeyError):
-                revmap_list = ["FREE MEMORY"]
-            yield (
-                0,
-                (
-                    str(string, "latin-1"),
-                    format_hints.Hex(offset),
-                    ", ".join(revmap_list),
-                ),
+
+            # We should really take care of this in the revmap generator
+            mapping_entry = revmap.get(
+                phys_offset & 0xFFFFFFFFFFFFF000, [("In Unallocated Space", 0)]
             )
+            for item in mapping_entry:
+                region, offset = item
+                if "Process" in region:
```

Comment:
String comparisons for differentiation seem horribly prone to typo issues in the future. Could this be turned into an enumeration or just something that contains the information other than in text form?

Reply:
same as above, its what comes from revmap generator, so would have to rewrite the revmapper